src/pipelinedefinitionstatus.hpp

```diff
@@ -303,6 +304,43 @@ class PipelineDefinitionStatus : public MachineState<BeginState, AvailableState,
                (state == PipelineDefinitionStateCode::LOADING_PRECONDITION_FAILED_REQUIRED_REVALIDATION) ||
                (state == PipelineDefinitionStateCode::BEGIN);
     }
+
+    std::tuple<ModelVersionState, ModelVersionStatusErrorCode> convertToModelStatus() const {
+        switch (getStateCode()) {
+        case PipelineDefinitionStateCode::BEGIN:
+            return {
+                ModelVersionState::LOADING,
+                ModelVersionStatusErrorCode::OK};
+
+        case PipelineDefinitionStateCode::LOADING_PRECONDITION_FAILED:
+            return {
+                ModelVersionState::LOADING,
+                ModelVersionStatusErrorCode::FAILED_PRECONDITION};
+
+        case PipelineDefinitionStateCode::LOADING_PRECONDITION_FAILED_REQUIRED_REVALIDATION:
+            return {
+                ModelVersionState::LOADING,
+                ModelVersionStatusErrorCode::OK};
+
+        case PipelineDefinitionStateCode::AVAILABLE_REQUIRED_REVALIDATION:
+            return {
```

Comment:
You could just have here:
case PipelineDefinitionStateCode::BEGIN:
case PipelineDefinitionStateCode::LOADING_PRECONDITION_FAILED_REQUIRED_REVALIDATION:
	            return {
                ModelVersionState::LOADING,
                ModelVersionStatusErrorCode::OK};
case PipelineDefinitionStateCode::AVAILABLE_REQUIRED_REVALIDATION:
case PipelineDefinitionStateCode::AVAILABLE:
return {
                ModelVersionState::AVAILABLE,
                ModelVersionStatusErrorCode::OK};

Which would point out more clearly that those states are treated as the same from ModelStatus point of view.

Reply:
Done